### PR TITLE
M6 modules, partest depends on scala as provided.

### PR DESCRIPTION
**Pull request by adriaanm**
The new partest now marks its scala dependencies as provided.
This is much more robust than before, and it breaks ugly cycles.

Updated the build for partest-extras to provide the required
scala-compiler dependency.

review by @gkossakowski

/cc @xeno-by: rebasing #3001 on this PR should fix it

### CODE REVIEW

**Comment by adriaanm** [comment]
Ah yes, I forgot this needs my buildcharacter.properties improvements for the sbt-for-ide dbuild.
Should have this passing on Monday.

**Comment by adriaanm** [comment]
Also, I think it's time to add `locker.skip=1` to `versions.properties`.

File: versions.properties [hunk 1]
```diff
@@ -1,7 +1,9 @@
-starr.version=2.11.0-M5
+starr.version=2.11.0-M6
+starr.use.released=1
```
Comment (gkossakowski) [inline versions.properties:2]: Does this imply skipping locker? If so, mention it in commit message or (preferably) make that change in separate commit.

It would be great to include some information why it's ok now to skip locker.

Comment (adriaanm) [inline versions.properties:2]: It doesn't imply skipping locker. That's `locker.skip`. `starr.use.released` means `starr.version` is used to resolve starr from maven. Otherwise, the old mechanism was used.

**Comment by adriaanm** [comment]
Failure is due to a maven error. Looks like maven is corrupting its own local cache (`/home/jenkins/.m2/` -- shared between all build nodes) on jenkins:

```
[artifact:dependencies] [INFO] snapshot org.scala-lang:scala-library:2.11.0-4d4503f-SNAPSHOT: checking for updates from extra-repo
[artifact:dependencies] [INFO] snapshot org.scala-lang:scala-reflect:2.11.0-4d4503f-SNAPSHOT: checking for updates from extra-repo
[artifact:dependencies] [WARNING] repository metadata for: 'snapshot org.scala-lang:scala-reflect:2.11.0-4d4503f-SNAPSHOT' could not be retrieved from repository: extra-repo due to an error: Error copying temporary file to the final destination: File /home/jenkins/.m2/repository/org/scala-lang/scala-reflect/2.11.0-4d4503f-SNAPSHOT/maven-metadata-extra-repo.xml.tmp does not exist
[artifact:dependencies] [INFO] Repository 'extra-repo' will be blacklisted
[artifact:dependencies] [WARNING] POM for 'org.scala-lang:scala-reflect:pom:2.11.0-4d4503f-SNAPSHOT:compile' is invalid.
```

**Comment by retronym** [comment]
AFAIK, maven local caches aren't multi-process safe and shouldn't be shared on the build server.

**Comment by adriaanm** [comment]
Yep, forgot about that. Luckily, we've solved that problem somewhere else. Just need to do it for ant-based maven tasks now.

**Comment by adriaanm** [comment]
I modified `~jenkins/.m2/settings.xml` as follows, so all maven resolution should now happen in isolation

```
<settings>
  <localRepository>${env.WORKSPACE}/m2repo</localRepository>
```

**Comment by adriaanm** [comment]
PLS REBUILD ALL

**Comment by adriaanm** [comment]
Urgh, the osgi tests failed because their custom maven repository manager does not support parametric localRepository paths. Or so it seems. UPDATED: true, it doesn't expand variables in localRepository, but the real failure is failing to do authentication on the moxie mirror (which it tries to do because it doesn't find the previously downloaded artifacts in the real local repo)

**Comment by adriaanm** [comment]
Fun times.

```
 Using manager SimpleLocalRepositoryManager with priority 0 for 
/localhome/jenkins/c/workspace/pr-scala-test/scala/${env.WORKSPACE}/m2repo
```

**Comment by adriaanm** [comment]
Oh man, the joys of code duplication. pax-url-aether is one shitty piece of software.
The version I upgraded to is no longer critically buggy, but it still doesn't expand variables in ~/.m2/settings.xml because they decided to reimplement their own parser for it instead of using Maven's.

https://ops4j1.jira.com/browse/PAXURL-138 -- they've had [PR](https://github.com/ops4j/org.ops4j.pax.url/pull/5) that supposedly fixes this for a year now.
https://ops4j1.jira.com/browse/PAXURL-217 -- fixed in 1.6.0

**Comment by adriaanm** [comment]
I'm merging this as soon as tests pass (fingers crossed!). Then, rebasing #3119, #3114, #3113, #3112, #3001 on top of master should result in happy PRs.

**Comment by adriaanm** [comment]
Result!

**Comment by adriaanm** [comment]
Sorry, @gkossakowski, I'm fast-tracking this because all these other PRs need to be rebased on this one and the only comment you had has been addressed.

**Comment by adriaanm** [comment]
I was wrong to rush this. Apologies. My oversight hopefully fixed in #3112's first commit, beed16825e.
